tests/framework/clients/rancher/client.go

```diff
@@ -68,9 +72,23 @@ func clientOpts(restConfig *rest.Config, rancherConfig *Config) *clientbase.Clie
 
 // GetRancherDynamicClient is a helper function that instantiates a dynamic client to communicate with the rancher host.
 func (c *Client) GetRancherDynamicClient() (dynamic.Interface, error) {
-	dynamic, err := frameworkDynamic.NewForConfig(c.session, c.restcConfig)
+	dynamic, err := frameworkDynamic.NewForConfig(c.Session, c.restConfig)
 	if err != nil {
 		return nil, err
 	}
 	return dynamic, nil
 }
+
+func (c *Client) GetManagementWatchInterface(resourseType string, opts metav1.ListOptions) (watch.Interface, error) {
```

Comment:
instead of asking for the resource string accept a runtime object and lookup the object in the Scheme.

Reply:
Look at new implementation. I did it this way with the schemaType which how the ops library works. All management.cattle.io resources have a string in their package the the ops library uses to find the schema.